### PR TITLE
refactor(console,core): remove add on dev feature guard

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Remove dev feature guard on @logto/core and @logto/console.

## Testing

N/A

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Review by gao-sun — APPROVED** [review]

File: packages/core/src/libraries/quota.ts [hunk 1]
```diff
@@ -1,154 +1,29 @@
...
-
-const shouldReportSubscriptionUpdates = (planId: string, key: keyof SubscriptionQuota): boolean =>
-  EnvSet.values.isDevFeaturesEnabled &&
+const shouldReportSubscriptionUpdates = (
+  planId: string,
+  key: keyof SubscriptionQuota,
+  isAddOnAvailable?: boolean
+): boolean =>
   planId === ReservedPlanId.Pro &&
+  Boolean(isAddOnAvailable) &&
```
Comment (wangsijie) [inline packages/core/src/libraries/quota.ts:23]: nit: this variable is already a boolean?

File: packages/core/src/middleware/koa-quota-guard.ts [hunk 2]
```diff
@@ -2,37 +2,17 @@ import { type Nullable } from '@silverhand/essentials';
...
 import { type QuotaLibrary } from '#src/libraries/quota.js';
-import { type SubscriptionQuota, type FeatureQuota } from '#src/utils/subscription/types.js';
+import { type SubscriptionQuota } from '#src/utils/subscription/types.js';
 
 type Method = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE' | 'COPY' | 'HEAD' | 'OPTIONS';
 
-/** @deprecated */
-type UsageGuardConfig = {
-  key: keyof FeatureQuota;
+type NewUsageGuardConfig = {
```
Comment (wangsijie) [inline packages/core/src/middleware/koa-quota-guard.ts:9]:
```suggestion
type UsageGuardConfig = {
```

File: packages/core/src/routes/sign-in-experience/custom-ui-assets/index.ts [hunk 3]
```diff
@@ -36,10 +36,8 @@ export default function customUiAssetsRoutes<T extends ManagementApiRouter>(
   router.post(
     '/sign-in-exp/default/custom-ui-assets',
     // Manually add this to avoid the case that the dev feature guard is removed but the quota guard is not being updated accordingly.
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    EnvSet.values.isDevFeaturesEnabled
-      ? newKoaQuotaGuard({ key: 'bringYourUiEnabled', quota })
-      : koaQuotaGuard({ key: 'bringYourUiEnabled', quota }),
+
+    newKoaQuotaGuard({ key: 'bringYourUiEnabled', quota }),
```
Comment (wangsijie) [inline packages/core/src/routes/sign-in-experience/custom-ui-assets/index.ts:40]: Consider name it `koaQuotaGuard`? Because the old one has been deleted

**Review by wangsijie — APPROVED** [review]
Overall LGTM for the core part.

**Pull request merged**